camel/toolkits/google_scholar_toolkit.py

```diff
@@ -77,21 +77,25 @@ def get_author_detailed_info(
         return author
 
     def get_author_publications(
-        self,
+        self, author: Optional[dict] = None
```

Comment:
how will user use this? they need manually set this dict?

Reply:
Author is is initialized through `get_author_detailed_info()`. After saving the author's information, the user can continue to the next step to obtain other data.

Reply:
I didn't get see benefit of the change

if user want to get publications from 2 authors, with current code:
```
author1_pub = GoogleScholarToolkit(author_identifier="id1").get_author_publications()

author2_pub = GoogleScholarToolkit(author_identifier="id2").get_author_publications()
``

with your code:
```
author1_pub = GoogleScholarToolkit(author_identifier="id1").get_author_publications()

author2_info = GoogleScholarToolkit(author_identifier="id2").get_author_detailed_info()
author2_pub = GoogleScholarToolkit(author_identifier="id2").get_author_publications(author2_info)
```

Reply:
Here is a possible scenario:
1. First get the author information, such as name, homepage, email_domain, and save it.
2. Then the next step is to determine whether a certain title is the user's publication.

I can only implement it like the following code:
```
author_info = GoogleScholarToolkit(author_identifier="id1").get_author_detailed_info()
save(author_info["name"], author_info["homepage"], author_info["email_domain"])
publications = author['publications']

for publication in publications:
    if publication['bib']['title'] == publication_title:
        return self.scholarly.fill(publication)
```

Or after saving the information, I want to get the title list:
```
author_info = GoogleScholarToolkit(author_identifier="id1").get_author_detailed_info()
save(author_info["name"], author_info["homepage"], author_info["email_domain"])
publications = author['publications']

publication_titles = [
    pub['bib']['title'] for pub in author['publications']
]
```

However, existing methods cannot be called.

Reply:
<img width="757" alt="Screenshot 2024-10-25 at 21 48 28" src="https://github.com/user-attachments/assets/3cc0037e-e676-46a2-b15f-e0b577a1bfcc">

what's this `publication_title`?

Reply:
and why we need  to determine whether a certain title is the user's publication

Reply:
> <img width="757" alt="Screenshot 2024-10-25 at 21 48 28" src="https://github.com/user-attachments/assets/3cc0037e-e676-46a2-b15f-e0b577a1bfcc">
> 
> what's this `publication_title`?

It's just a hypothesis that if there is a need, how do you do it?